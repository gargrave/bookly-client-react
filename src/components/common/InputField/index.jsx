// @flow
import React from 'react';
import { func, oneOf, string } from 'prop-types';

type Props = {
  boundValue: string,
  label: string,
  name: string,
  onInputChange: Function,
  type?: string,
};

const acceptableTypes = [
  'email',
  'password',
  'text',
];

function InputField({
  boundValue,
  label,
  name,
  onInputChange,
  type,
}: Props) {
  return (
    <div className="input-field">
      <label htmlFor={name}>{label}:</label>
      <input
        type={type || 'text'}
        name={name}
        id={name}
        value={boundValue}
        onChange={onInputChange} />
    </div>
  );
}

InputField.propTypes = {
  boundValue: string.isRequired,
  label: string.isRequired,
  name: string.isRequired,
  onInputChange: func.isRequired,
  type: oneOf(acceptableTypes),
};

export default InputField;
