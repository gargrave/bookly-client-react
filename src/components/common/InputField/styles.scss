.bookly-input-field {
  &__input {
    &--invalid {
      border-color: #ff4136 !important; // using !important to override Shoelace's styles
    }
  }

  &__error {
    color: #ff4136;
  }
}
