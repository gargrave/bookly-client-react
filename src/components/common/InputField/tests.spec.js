import React from 'react';
import { shallow } from 'enzyme';

import InputField from './';

describe('InputField', () => {
  let props;
  let component;


  describe('"text" type input field', () => {
    beforeEach(() => {
      props = {
        boundValue: '',
        label: 'Whatever',
        name: 'text',
        onInputChange: jest.fn(),
        type: 'text',
      };

      component = shallow(<InputField {...props} />);
    });

    test('renders a text input correctly', () => {
      expect(component).toMatchSnapshot();
      expect(component.find('input[type="email"]').length).toEqual(0);
      expect(component.find('input[type="text"]').length).toEqual(1);
      expect(component.find('input[type="password"]').length).toEqual(0);
    });
  });

  describe('"password" type input field', () => {
    beforeEach(() => {
      props = {
        boundValue: '',
        label: 'Password',
        name: 'password',
        onInputChange: jest.fn(),
        type: 'password',
      };

      component = shallow(<InputField {...props} />);
    });

    test('renders a password input correctly', () => {
      expect(component).toMatchSnapshot();
      expect(component.find('input[type="email"]').length).toEqual(0);
      expect(component.find('input[type="text"]').length).toEqual(0);
      expect(component.find('input[type="password"]').length).toEqual(1);
    });
  });

  describe('"email" type input field', () => {
    beforeEach(() => {
      props = {
        boundValue: '',
        label: 'Email',
        name: 'email',
        onInputChange: jest.fn(),
        type: 'email',
      };

      component = shallow(<InputField {...props} />);
    });

    test('renders an email input correctly', () => {
      expect(component).toMatchSnapshot();
      expect(component.find('input[type="email"]').length).toEqual(1);
      expect(component.find('input[type="password"]').length).toEqual(0);
      expect(component.find('input[type="text"]').length).toEqual(0);
    });
  });
});
