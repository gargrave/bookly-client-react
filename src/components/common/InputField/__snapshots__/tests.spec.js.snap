// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField "email" type input field renders an email input correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    boundValue=""
    label="Email"
    name="email"
    onInputChange={[Function]}
    type="email"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="email"
        >
          Email
          :
        </label>,
        <input
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />,
      ],
      "className": "input-field",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Email",
            ":",
          ],
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": Array [
          "Email",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "name": "email",
          "onChange": [Function],
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="email"
          >
            Email
            :
          </label>,
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />,
        ],
        "className": "input-field",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email",
              ":",
            ],
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": Array [
            "Email",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "name": "email",
            "onChange": [Function],
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InputField "password" type input field renders a password input correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    boundValue=""
    label="Password"
    name="password"
    onInputChange={[Function]}
    type="password"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="password"
        >
          Password
          :
        </label>,
        <input
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />,
      ],
      "className": "input-field",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Password",
            ":",
          ],
          "htmlFor": "password",
        },
        "ref": null,
        "rendered": Array [
          "Password",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "name": "password",
          "onChange": [Function],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="password"
          >
            Password
            :
          </label>,
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />,
        ],
        "className": "input-field",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Password",
              ":",
            ],
            "htmlFor": "password",
          },
          "ref": null,
          "rendered": Array [
            "Password",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "name": "password",
            "onChange": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InputField "text" type input field renders a text input correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    boundValue=""
    label="Whatever"
    name="text"
    onInputChange={[Function]}
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="text"
        >
          Whatever
          :
        </label>,
        <input
          id="text"
          name="text"
          onChange={[Function]}
          type="text"
          value=""
        />,
      ],
      "className": "input-field",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Whatever",
            ":",
          ],
          "htmlFor": "text",
        },
        "ref": null,
        "rendered": Array [
          "Whatever",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "text",
          "name": "text",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="text"
          >
            Whatever
            :
          </label>,
          <input
            id="text"
            name="text"
            onChange={[Function]}
            type="text"
            value=""
          />,
        ],
        "className": "input-field",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Whatever",
              ":",
            ],
            "htmlFor": "text",
          },
          "ref": null,
          "rendered": Array [
            "Whatever",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "text",
            "name": "text",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
